godot-codegen/src/util.rs

```diff
@@ -55,6 +55,34 @@ pub fn make_sname_ptr(identifier: &str) -> TokenStream {
     }
 }
 
+// Unclear on if some of these classes should be registered earlier than `Scene`:
+// - `RenderData` + `RenderDataExtension`
+// - `RenderSceneData` + `RenderSceneDataExtension`
+
+#[rustfmt::skip]
+fn class_api_level_is_server(class_name: &str) -> bool {
+    matches!(class_name, 
+        // TODO: These should actually be at level `Core`
+        "Object" | "OpenXRExtensionWrapperExtension" |
+
+        // Shouldn't be inherited from in rust but are still servers.
+        "AudioServer" | "CameraServer" | "NavigationServer2D" | "NavigationServer3D" | "RenderingServer" | "TranslationServer" | "XRServer" |
+
+        // PhysicsServer2D
+        "PhysicsDirectBodyState2D" | "PhysicsDirectBodyState2DExtension" |
+        "PhysicsDirectSpaceState2D" | "PhysicsDirectSpaceState2DExtension" | 
+        "PhysicsServer2D" | "PhysicsServer2DExtension" |
+        "PhysicsServer2DManager" |
+
+        // PhysicsServer3D
+        "PhysicsDirectBodyState3D" | "PhysicsDirectBodyState3DExtension" |
+        "PhysicsDirectSpaceState3D" | "PhysicsDirectSpaceState3DExtension" | 
+        "PhysicsServer3D" | "PhysicsServer3DExtension" | 
+        "PhysicsServer3DManager" |
+        "PhysicsServer3DRenderingServerHandler" 
+    )
+}
+
```

Comment:
Thanks for listing them all! :+1: 

What do you think about moving this to `special_cases.rs`, where we already have lots of hardcoded lists?

The naming could then also be consistent with `is_class_experimental`/`is_class_deleted`, e.g. `is_class_level_server` or so.

Maybe it could then also use `match` + `=> true, _ => false` like the others, which tends to work a bit nicer with syntax highlighting and IDEs than the macro.